### PR TITLE
handle expiring/ed ds embargoes where the ds does not exist

### PR DESCRIPTION
**Pull request by nhart**
**JIRA Ticket**: https://jira.lyrasis.org/browse/ISLANDORA-2528

# What does this Pull Request do?

The module allows embargos for datastreams that don't exist; lets take the same approach when lifting embargoes.

# What's new?
Don't require the object to have the referenced datastream in the embargo. If it doesn't exist, construct it and send it along.

# How should this be tested?
Set a scholar embargo for a datastream that doesn't exist.
Let it expire
Confirm cron has run
Confirm the RELS-INT and POLICY datastreams no longer reference the datastream that was embargoed.

# Interested parties
@Islandora/7-x-1-x-committers

### CODE REVIEW

File: modules/islandora_scholar_embargo/islandora_scholar_embargo.module
Comment (adam-vessey) [inline modules/islandora_scholar_embargo/islandora_scholar_embargo.module]: Not sure this is valid... it's expecting an `islandora_datastream`: https://github.com/Islandora/islandora_scholar/blob/ab835969c995be48bb5d3e44268bcb37c686a008/modules/islandora_scholar_embargo/islandora_scholar_embargo.rules.inc#L72

File: modules/islandora_scholar_embargo/islandora_scholar_embargo.module
Comment (adam-vessey) [inline modules/islandora_scholar_embargo/islandora_scholar_embargo.module]: Same deal, this is expecting an `islandora_datastream`, not what is probably a string?: https://github.com/Islandora/islandora_scholar/blob/ab835969c995be48bb5d3e44268bcb37c686a008/modules/islandora_scholar_embargo/islandora_scholar_embargo.rules.inc#L40

**Review by IAMlKeno — APPROVED** [review]

**Comment by nhart** [comment]
Any @Islandora/7-x-1-x-committers  free to look at this?

**Comment by whikloj** [comment]
@nhart I'm not a Scholar user, but if I understand.
1. Create some resource, and add an embargo on a datastream that does not exist. This is currently allowed. 
2. When the embargo is lifted it would throw some sort of exception.

After this PR, step 2 above would not have an error.

**Comment by bondjimbond** [comment]
I'm not sure why `$item->constructDatastream($params['dsid']);` is the way it works, but it seems to be OK.

Are we ignoring Travis at this point? It is reporting errors that have nothing to do with this PR:

```
sites/all/modules/islandora_scholar/includes/page_callbacks.inc:
 +47: [minor] Line exceeds 80 characters; contains 104 characters
 +48: [minor] Line exceeds 80 characters; contains 85 characters
 +67: [minor] Line exceeds 80 characters; contains 98 characters
 +122: [minor] Expected 1 space before "="; 0 found
 +227: [minor] Function comment short description must end with a full stop
 +229: [minor] Missing comment for param "$search_query" at position 1
 +230: [minor] Missing comment for param "$tag" at position 2
 +232: [minor] Missing comment for @return statement
 +274: [minor] Functions must not contain multiple empty lines in a row; found 2 empty lines
sites/all/modules/islandora_scholar/includes/upload.tab.inc:
 +225: [minor] Line exceeds 80 characters; contains 103 characters
 +229: [minor] AS keyword must be lowercase; expected "as" but found "AS"
 +230: [minor] There should be no white space after an opening "("
 +238: [minor] Functions must not contain multiple empty lines in a row; found 2 empty lines
sites/all/modules/islandora_scholar/islandora_scholar.install:
 +155: [minor] Function comment short description must end with a full stop
 +155: [minor] Line exceeds 80 characters; contains 107 characters
 +169: [minor] Files must end in a single new line character
```

**Comment by nhart** [comment]
@whikloj  Close.. The expired embargo was never lifted due to this [check](https://github.com/Islandora/islandora_scholar/blob/7.x/modules/islandora_scholar_embargo/islandora_scholar_embargo.module#L314); only lifting when the object had the datastream.

**Comment by whikloj** [comment]
I think I understand now. This makes sense to me. I'm :+1: to merge this if
you're happy @bondjimbond.

**Comment by bondjimbond** [comment]
I'm OK with it @whikloj

**Review by whikloj — APPROVED** [review]
👍

**Pull request merged**
